src/boot/bootctl.c

```diff
@@ -1723,7 +1724,15 @@ static int verb_status(int argc, char *argv[], void *userdata) {
                 printf("  Secure Boot: %sd (%s)\n",
                        enable_disable(IN_SET(secure, SECURE_BOOT_USER, SECURE_BOOT_DEPLOYED)),
                        secure_boot_mode_to_string(secure));
-                printf(" TPM2 Support: %s\n", yes_no(efi_has_tpm2()));
+
+                s = tpm2_support();
+                printf(" TPM2 Support: %s%s%s\n",
+                       FLAGS_SET(s, TPM2_SUPPORT_FIRMWARE|TPM2_SUPPORT_DRIVER) ? ansi_highlight_green() :
+                       (s & (TPM2_SUPPORT_FIRMWARE|TPM2_SUPPORT_DRIVER)) != 0 ? ansi_highlight_red() : ansi_highlight_yellow(),
```

Comment:
I think you have yellow and red reversed here or this should be `==`…

Reply:
nope, this was done on purpose. i.e. having partial tpm support is a weird thing. having no tpm support is quite ok. hence i highlighted the partial (i.e. kinda broken) support as red, but the tpm being absent just as yellow, because it is noteworthy but kinda ok if your platform doesn't have it.